pyanaconda/modules/storage/nvdimm/nvdimm.py

```diff
@@ -46,3 +54,46 @@ def process_kickstart(self, data):
     def setup_kickstart(self, data):
         """Setup the kickstart data."""
         data.nvdimm.actionList = self._actions
+
+    def get_namespaces_to_use(self):
+        """Get namespaces to be used.
+
+        FIXME: Can we return an empty string in the set?
+
+        :return: a set of namespaces
+        """
+        return {
+            action.namespace for action in self._actions
+            if action.action == NVDIMM_ACTION_RECONFIGURE
+            or (action.action == NVDIMM_ACTION_USE and action.namespace)
+        }
+
+    def get_devices_to_use(self):
+        """Get devices to be used.
+
+        :return: a set to device names
+        """
+        return {
+            dev for action in self._actions for dev in action.blockdevs
+            if action.action == NVDIMM_ACTION_USE and action.blockdevs
+        }
+
+    def get_devices_to_ignore(self):
+        """Get devices to be ignored.
+
+        By default nvdimm devices are ignored. To become available for installation,
+        the device(s) must be specified by nvdimm kickstart command.
+        Also, only devices in sector mode are allowed.
+
+        :return: a set of device names
+        """
+        namespaces = self.get_namespaces_to_use()
+        devices = self.get_devices_to_use()
+
+        return {
+            ns_info.blockdev for ns_name, ns_info in nvdimm.namespaces.items()
+            if ns_info.mode != blockdev.NVDIMMNamespaceMode.SECTOR
+            and ns_info.blockdev
+            and ns_name not in namespaces
+            and ns_info.blockdev not in devices
```

Comment:
This implementation looks nice but we are loosing logging messages which could be important, don't we?

Reply:
Not really, the result is logged in `ignore_nvdimm_blockdevs`.

Reply:
I see there information that the devices were added but I'm still missing reason why they are ignored when they are ignored. Am I missing something?

Reply:
Fixed.